### PR TITLE
Pre-install_check: Update Pre-Install Check based newer Viya4 Requirements Documentation (ls_issue_79)

### PR DESCRIPTION
**Pull request by lasiva**
1. Dropped code to validate individual nodes Memory and CPU for all nodes in cluster.  Tool only checks for Aggregates capacity memory and capacity CPU of all nodes in the cluster.  This dropped following keys in viya_deployment_settings.ini which resulted in refactoring interfaces:
# Minimum CPU of User node
VIYA_GENERIC_WORKER_CPU=8
# Minimum Memory of User node expressed in Gi or Ki. Minimum value > 0 Gi/Ki
VIYA_GENERIC_WORKER_MEMORY=32Gi

2. Aggregate Memory and CPU are based on capacity of the node,  the Allocatable memory or CPU.  This resulted on refactoring a lot of variable names.  It is based on capacity because kubectl cannot determine instance type.
3. viya_cluster_settings.properties  file replaced with viya_deployment_settings.ini which uses configparser

Please check README files for language and accuracy.
Please check  viya_deployment_settings.ini for consistency with README.
Most code change are related to refactoring variable names.

### CODE REVIEW

**Comment by FredPerrySAS** [comment]
The "Memory and vCPU Check" header needs two or three hash marks to make it a header. I think that will fix the spacing problem in the section that follows.

Also, the following paragraph includes "SAS VIYA Operations document". Viya should not be in all caps.

**Comment by AlissaSchleich** [comment]
I looked over the readme and looked over the changes in the .ini files and they look good to me

**Review by AlissaSchleich — APPROVED** [review]
I read over the readme and looked over the ini changes.   All looks okay to me.

**Review by FredPerrySAS — CHANGES_REQUESTED** [review]
In the README, the "Memory and vCPU Check" header needs two or three hash marks to make it a header. I think that will fix the spacing problem in the content that follows.

Also, the paragraph that follows includes "SAS VIYA Operations document". Viya should not be in all caps.

**Review by FredPerrySAS — APPROVED** [review]
Changes look good. Thank you.

File: pre_install_report/README.md [hunk 1]
```diff
@@ -1,15 +1,60 @@
 # Pre-Installation Check of SAS Viya System Requirements
+## Overview
+This tool compares your Kubernetes environment to the SAS Viya system requirements.  It evaluates a   
+number of items, such as memory, CPU cores, software versions, and permissions. The output is a web-viewable,  
+HTML report with the results. SAS recommends running the tool and resolving any reported issues before   
+beginning a SAS Viya deployment in a Kubernetes cluster.  
+
+### Memory and vCPU Check
+The tool calculates the aggregate Memory and aggregate vCPUs of all the nodes that must be active and running. The aggregate
+Memory is the sum the Memory capacity on all the running nodes. The aggregate CPU is calculated similarly.
```
Comment (kevinlinglesas) [inline pre_install_report/README.md:10]: Should this be "sum of the Memory"?

File: pre_install_report/README.md [hunk 2]
```diff
@@ -1,15 +1,60 @@
...
+VIYA_MIN_AGGREGATE_WORKER_MEMORY=448Gi
+# Total CPU of all worker Nodes in millicores.  Sum of the vCPUs on all active node required to deploy a specific offering.
+# Minimum allowed value = '.001'. 
+VIYA_MIN_AGGREGATE_WORKER_CPU_CORES=56
+```
+
+If calculated aggregate memory is less than VIYA_MIN_AGGREGATE_WORKER_MEMORY then the tool will flag a memory issue.  
+If calculated aggregate vCPUs is less than VIYA_MIN_AGGREGATE_WORKER_CPU_CORES then it the tool will flag a CPU issue.
+ 
+SAS recommends using the SAS Viya 4 Infrastructure as Code (IaC) tools to create a cluster for Microsoft Azure, AWS,
```
Comment (kevinlinglesas) [inline pre_install_report/README.md:31]: Can you embed the links to the IaC GitHub repos like so... 
"SAS recommends using SAS Viya 4 Infrastructure as Code (IaC) tools to create a cluster for [Microsoft Azure](https://github.com/sassoftware/viya4-iac-azure), [AWS](https://github.com/sassoftware/viya4-iac-aws), or [GCP](https://github.com/sassoftware/viya4-iac-gcp)"

File: pre_install_report/README.md (hunk 1 above)
Comment (lasiva) [reply pre_install_report/README.md:10]: Will modify.

File: pre_install_report/README.md (hunk 2 above)
Comment (lasiva) [reply pre_install_report/README.md:38]: If I copy this as is I don't get the links in the README.

Comment (lasiva) [reply pre_install_report/README.md:38]: Making needed changes.  Please review after push

File: pre_install_report/library/pre_install_check.py [hunk 3]
```diff
@@ -710,27 +706,22 @@ def evaluate_nodes(self, nodes_data, global_data, cluster_info, quantity_):
...
-            # Node is identified as worker node
+            capacity_cpu_cores = self._get_cpu_units(node, 'cpu')
+            # alloc_cpu_cores = self._get_cpu_units(node, 'allocatablecpu')
+
             kubeletversion = str(node['kubeletversion'])
-            alloc_memory = str(node['allocatableMemory'])
-            total_memory = total_memory + quantity_(str(node['memory']))
-            total_allocatable_memory = total_allocatable_memory + quantity_(alloc_memory)
+            capacity_memory = str(node['memory'])
+            # allocatable_memory = str(node['allocatableMemory'])
```
Comment (screnshaw) [inline pre_install_report/library/pre_install_check.py:723]: You can delete the comment at 723

File: pre_install_report/viya_deployment_settings.ini [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+[items]
+# TOTAL Memory of all worker Nodes in GB. Sum of the Memory on all active node required to deploy a specific offering.
+VIYA_MIN_AGGREGATE_WORKER_MEMORY=104G
+# TOTAL CPU of all worker Nodes in millicores. Sum of the vCPUs on all active node required to deploy a specific offering
+# Minimum allowed value = '.001'.
+VIYA_MIN_AGGREGATE_WORKER_CPU_CORES=22
+# Supported versions of Kubelet Version
+# Expected: xx if used for Viya Deployment
+VIYA_KUBELET_VERSION_MIN=v1.19.0
+
```
Comment (screnshaw) [inline pre_install_report/viya_deployment_settings.ini:34]: Change the defaults back to your original settings for standard deployment.

**Review by screnshaw — APPROVED** [review]
Reviewed with comments.

File: pre_install_report/library/pre_install_check.py (hunk 3 above)
Comment (lasiva) [reply pre_install_report/library/pre_install_check.py:723]: Done

File: pre_install_report/README.md [hunk 5]
```diff
@@ -1,15 +1,67 @@
...
+
+
+| Offering                  | CAS Node(s)          | System Node  | Nodes in User Node Pool(s)  |
+| ------------------------- |-------------          | --------- | -------------|
+| SAS Visual Analytics and SAS Data Preparation  |  Num of Node: Node 1, CPU: 16, Memory, 128 | Num. of Nodes: 1, CPU: 8,  Memory: 64 | Num of Node: 1 per Node User Node Pool,  CPU: 8 Memory 64 |
 
-This tool compares your Kubernetes environment to the SAS Viya system requirements. It evaluates a number of items, such as memory, CPU cores, software versions, and permissions. The output is a web-viewable, HTML report with the results. 
+VIYA_MIN_AGGREGATE_WORKER_MEMORY =  
+CAS Node Memory + System Node Memory + (User Nodes Memory * 4)  
+fl
```
Comment (kevinlinglesas) [inline pre_install_report/README.md:52]: I don't think this "fl" is supposed to be there.

**Pull request merged**